judge/models/contest.py

```diff
@@ -182,17 +186,32 @@ def can_see_scoreboard(self, user):
             return False
         if self.hide_scoreboard and not self.is_in_contest(user) and self.end_time > self._now:
             return False
+        if self.partially_hide_scoreboard and not self.is_finished_contest(user):
```

Comment:
This is 2 queries (one in `can_see_scoreboard` and one in `can_see_full_scoreboard`) per call to `can_see_scoreboard`. Can you reduce it to one?

Reply:
I don't see a good way to reduce it down to one query `self.hide_scoreboard` has the same problem.